builtin/providers/github/resource_github_membership_test.go

```diff
@@ -0,0 +1,84 @@
+package github
+
+import (
+	"fmt"
+	"github.com/google/go-github/github"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"testing"
+)
+
+func TestAccGithubMembership_basic(t *testing.T) {
+	var membership github.Membership
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGithubMembershipDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccGithubMembershipConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGithubMembershipExists("github_membership.test_org_membership", &membership),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGithubMembershipDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*GithubClient).client
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "github_membership" {
+			continue
+		}
+		o, u := parseMembershipId(rs.Primary.ID)
+
+		membership, resp, err := conn.Organizations.GetOrgMembership(u, o)
+
+		if err == nil {
+			memName := membership.Organization.Login
+			userName := membership.User.Login
+			if membership != nil &&
+				buildMembershipId(*memName, *userName) == rs.Primary.ID {
+				return fmt.Errorf("Organization membership still exists")
+			}
+		}
+		if resp.StatusCode != 404 {
+			return err
+		}
+		return nil
+	}
+	return nil
+}
+
+func testAccCheckGithubMembershipExists(n string, membership *github.Membership) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not Found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Team ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*GithubClient).client
+		o, u := parseMembershipId(rs.Primary.ID)
+
+		githubMembership, _, err := conn.Organizations.GetOrgMembership(u, o)
+		if err != nil {
+			return err
+		}
+		*membership = *githubMembership
+		return nil
+	}
+}
```

Comment:
I suspect it will involved adding another spec like `TestAccGithubMembership_invalidRole` that uses a different `const` config with an invalid value. Not sure how to test it using their framework though, might need to dig to find an example in their codebase.